app/javascript/packs/puerto-rico-guidance.ts

```diff
@@ -13,5 +17,30 @@ function onStateSelectionChange() {
   showOrHidePuertoRicoExtras(stateSelector?.value);
 }
 
+function onIdentityDocStateSelection() {
+  const stateSelectors = document.querySelectorAll<HTMLSelectElement>('.usa-select');
+  stateSelectors.forEach((stateSelector) => {
+    if (!stateSelector.id.includes('jurisdiction')) {
+      stateSelector.addEventListener('change', () =>
+        showOrHidePuertoRicoExtras(stateSelector.value),
+      );
+      showOrHidePuertoRicoExtras(stateSelector.value);
+    }
+  });
+}
+
+function onUpdate() {
+  const updateStateSelectors = document.querySelectorAll<HTMLSelectElement>('.usa-select');
```

Comment:
Since `onUpdate` and `onIdentityDocStateSelection` only run during initial load, I think it makes sense to combine the functions and inner loops to decrease the complexity and runtime here. DOM traversal in particular is on the expensive side of things when it comes to runtime.